Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Motion/MotionDataBuilder.cpp

```diff
@@ -166,6 +165,35 @@ namespace EMotionFX
             return finalMotionData;
         }
 
+        AZ::SceneAPI::DataTypes::MatrixType MotionDataBuilder::GetLocalSpaceBindPose(const SceneContainers::SceneGraph& sceneGraph,
+            const SceneContainers::SceneGraph::NodeIndex rootBoneNodeIndex,
+            const SceneContainers::SceneGraph::NodeIndex nodeIndex,
+            const SceneDataTypes::ITransform* transform,
+            const SceneDataTypes::IBoneData* bone) const
+        {
+            if (nodeIndex != rootBoneNodeIndex)
+            {
+                const SceneContainers::SceneGraph::NodeIndex parentNodeIndex = sceneGraph.GetNodeParent(nodeIndex);
+                const AZStd::shared_ptr<const SceneDataTypes::IGraphObject> parentNode = sceneGraph.GetNodeContent(parentNodeIndex);
+                const AZStd::shared_ptr<const SceneDataTypes::IBoneData> parentBone = azrtti_cast<const SceneDataTypes::IBoneData*>(parentNode);
+                if (parentBone)
```

Comment:
This creates a copy of the `shared_ptr` instance, incrementing the use count unnecessarily. This would be better written like:
```suggestion
                if (const SceneDataTypes::IBoneData* parentBone = azrtti_cast<const SceneDataTypes::IBoneData*>(parentNode));
```
Putting the declaration of `parentBone` in the `if` statement tightens the scope of the variable, so it is only declared and usable when the variable is set to a valid value.

Reply:
Changed